host-spec/c04-networking.tm

```diff
@@ -99,533 +248,371 @@
+    header hash>|<cell|Sect. <reference|sect-blake2>>>|<row|<cell|bytes>|<cell|2>|<cell|Block
+    header (optional)>|<cell|Def. <reference|block>>>|<row|<cell|repeated>|<cell|3>|<cell|Block
+    body (optional)>|<cell|Def. <reference|sect-block-body>>>|<row|<cell|bytes>|<cell|>|<cell|>|<cell|>>|<row|<cell|bytes>|<cell|4>|<cell|Block
+    receipt (optional)>|<cell|>>|<row|<cell|bytes>|<cell|5>|<cell|Block
+    message queue (optional)>|<cell|>>|<row|<cell|bytes>|<cell|6>|<cell|Justification
+    (optional)>|<cell|Def. <reference|defn-grandpa-justification>>>|<row|<cell|bool>|<cell|7>|<cell|Indicates
+    whether the justification>|<cell|>>|<row|<cell|>|<cell|>|<cell|is empty
+    (i.e. should be ignored).>|<cell|>>>>>>
+      <strong|BlockData> Protobuf structure.
+    </big-table>
+  </definition>
+
+  <subsubsection|Transactions><label|sect-msg-transactions>
+
+  Transactions as defined and described in Section
+  <reference|sect-transactions> are sent directly to peers with which the
+  Polkadot Host has an open transaction substream, as defined in Definition
+  <reference|defn-transactions-message>. Polkadot Host implementers should
+  implement a mechanism which only sends a transaction once to each peer and
+  avoids sending duplicates. Sending duplicate transactions might result in
+  undefined consequences such as being blocked for bad behavior by peers.
+
+  The mechanism for managing transactions is further described in Section
+  <reference|sect-extrinsics> respectively Section
+  <reference|sect-transaction-submission> and Section
+  <reference|sect-transaction-queue>.
+
+  <\definition>
+    <label|defn-transactions-message>The <strong|transactions message> is the
+    structure of how the transactions are sent over the network. It is
+    represented by <math|M<rsub|T>> and is defined as follows:
+
+    <\equation*>
+      M<rsub|T>\<assign\>Enc<rsub|SC><around*|(|C<rsub|1>,\<ldots\>,C<rsub|n>|)>
+    </equation*>
+
+    in which:
+
+    <\equation*>
+      C<rsub|i>\<assign\>Enc<rsub|SC><around*|(|E<rsub|i>|)>
+    </equation*>
+
+    Where each <math|E<rsub|i>> is a byte array and represents a sepearate
+    extrinsic. The Polkadot Host is agnostic about the content of an
+    extrinsic and treats it as a blob of data.
+
+    Transactions are sent over the <verbatim|/dot/transactions/1> substream.
+  </definition>
+
+  <subsubsection|GRANDPA Votes><label|sect-msg-grandpa>
+
+  The exchange of GRANDPA messages is conducted on the
+  <verbatim|/paritytech/grandpa/1> substream. The process for the creation of
+  such votes is described in Section <reference|sect-finality>.
```

Comment:
Can we just move Grandpa's sections on Network encoding here?

Reply:
Yes, we probably should. I will do that in this issues, however: #369